### PR TITLE
Add Geo3K Tinker training example with VLM support

### PR DESCRIPTION
**Pull request by BrianChen26**
V1: 
- Add examples/geo3k_tinker/ with training entrypoint, shell scripts, and inference scripts
- Add VLM support to TinkerEngine (image_processor parameter)
- Add VLM detection and image_processor loading in TinkerWorkflowTrainer
- Update model to Qwen/Qwen3-VL-30B-A3B-Instruct (Tinker-supported)

Changes

- rllm/engine/rollout/tinker_engine.py: Added optional `image_processor` parameter to support Vision-Language Models (VLMs) by passing it to Tinker's renderer system.

- rllm/trainer/tinker/tinker_workflow_trainer.py: Added automatic VLM detection and `image_processor` loading for models containing "vl" or "vision" in their name. Backward compatible with non-VLM models.

- examples/geo3k_tinker/train_geo3k_tinker.py: New training entrypoint for Geo3K workflow using Tinker backend, mirroring the Verl version but with `backend="tinker"`.

- examples/geo3k_tinker/run_geo3k_tinker.py: New inference/evaluation script using `TinkerEngine` instead of `OpenAIEngine`, saves results to `logs/geo3k_tinker.json`.

- examples/geo3k_tinker/train_geo3k_tinker.sh: Training shell script that configures model name, batch sizes, and enables wandb logging for Geo3K Tinker training.

Testing: 
- Ran 100 training steps, with accuracy around 0.13 to 0.16
 
V2: 
Fixed issue preventing VLMs from correctly processing images during Geo3K training with Tinker backend. Validation accuracy improved from ~16% to ~55%.

Problem:
- The VLM (Qwen/Qwen3-VL-30B-A3B-Instruct) was not processing images correctly, resulting in the model training on text-only prompts. Root causes:
- image_processor was not being passed to the Tinker renderer
- Message format was incompatible with Qwen3-VL expectations
- to_ints() raised ValueError for prompts containing ImageChunks
- Assertion failed for empty prompt_ids (valid for VLM prompts)

Changes:
- rllm/engine/rollout/tinker_engine.py: Added image_processor parameter and passed it to the Tinker renderer; added try-except to handle ValueError from to_ints() when prompts contain ImageChunks.
- rllm/trainer/tinker/tinker_workflow_trainer.py: Added auto-detection and loading of image_processor for VL models using AutoProcessor; fixed assertion to use is not None instead of truthiness check to allow empty prompt_ids lists.
- examples/geo3k/geo3k_workflow.py: Updated message format to Qwen3-VL compatible format with {"type": "image", "image": <PIL.Image>} content arrays.
- examples/geo3k_tinker/run_geo3k_tinker.py: Added image_processor loading and passing to TinkerEngine for inference script.
- examples/geo3k_tinker/train_geo3k_tinker.py: No functional changes (uses workflow trainer which now handles image_processor).

Testing:
- Ran 100 training steps, accuracy around 0.53 - 0.6

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -133,7 +133,7 @@ flash-attn = [
 ]
 
 [tool.ruff]
-line-length = 5000 # TODO: Reduce this to a more reasonable value
+line-length = 320 # TODO: Reduce this to a more reasonable value
```
Comment (jeffreysijuntan) [inline pyproject.toml]: Can you revert this back to 5000, and rerun `pre-commit run --all-files`? Right now the pre-commit check is still failing.

Comment (BrianChen26) [reply pyproject.toml]: Fixed, pre-commit run --all-files passes now.
